algorithms/src/Kokkos_Random.hpp

```diff
@@ -687,6 +687,18 @@ struct Random_UniqueIndex<Kokkos::Experimental::SYCL> {
 };
 #endif
 
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+template <>
+struct Random_UniqueIndex<Kokkos::Experimental::OpenMPTarget> {
+  using locks_view_type = View<int*, Kokkos::Experimental::OpenMPTarget>;
+  KOKKOS_FUNCTION
+  static int get_state_idx(const locks_view_type& locks) {
+    const int i = omp_get_team_num() * 1024 + omp_get_thread_num();
+    return i % locks.extent(0);
```

Comment:
Why don't you need a spin loop here as we have for the other backends?

Reply:
because me being stupid. You are obviously right thats just he guess and then the while loop needs to be there DUH